text/rocq-roadmap.md

```diff
@@ -5,7 +5,7 @@ Rocq Long Term Roadmap
 
 The Rocq Development Team, building on 40 years of experience with Coq, is pleased to announce the Rocq Prover, the next-generation proof assistant. The Rocq Prover is designed to empower a diverse range of users, from mathematicians seeking formal rigor to software engineers building high-assurance systems.
 
-While Rocq leverages the rich heritage of Coq, it embraces a more agile development philosophy. This allows us to deliver impactful changes more frequently, accelerating innovation and user experience improvements.
+Rocq builds upon the rich heritage and diverse ecosystem of Coq. We promote a development methodology which achieves the delicate balance of being mindful of compatibility and ecosystem maintainability, while sufficiently agile to deliver impactful changes frequently, boosting research and innovation, and continuously improving user experience.
```

Comment:
For some reason, I can't use the suggestion widget on this text :-(.

> We promote a development methodology which achieves the delicate balance of being mindful of compatibility and ecosystem maintainability

A lot of vague words.  Maybe something like this, but later in the document:

"We aim to maintain backward compatibility for existing users to a large degree without unduly limiting our ability to deliver improvements.  You may need to make limited changes to migrate your project to future versions of Rocq."

> boosting research and innovation,

This is a top-level goal, it should be in its own sentence, perhaps in the first paragraph.  Maybe something like

"The Rocq Development Team works to support improved proof verification techniques, to promote greater use and understanding of proof verification and to support research in topics related to proof verification"

> while sufficiently agile to deliver impactful changes frequently,
> and continuously improving user experience.

These could apply to almost any software project and therefore they're not interesting.  They would be a lot more interesting if they provided new information tied to the specifics of Rocq.  But they probably shouldn't be in this position in the document.

Reply:
> > These could apply to almost any software project and therefore they're not interesting.
> 
> That's not true, though. Lots of software aims for strong backward-compatibility at the detriment of fast-paced evolution.

My comment was about this text rather than backward compatibility:

> while sufficiently agile to deliver impactful changes frequently,
and continuously improving user experience.

"agile" is vague.  Maybe something about frequent releases would be better
"frequently" is in part (maybe in large part) a question of resources
"continuously improving user experience" - this seems like an aspirational goal.  At the least, I'm not sure what recent work I'd put under "improved user experience"

Reply:
> At the least, I'm not sure what recent work I'd put under "improved user experience"

Tons of work has recently been done for improved user experience. In fact, the part about usability in this document is becoming too long, precisely because there is so much going on there (most of the items listed are not just aspirational but correspond to real ongoing work).

Reply:
"User experience" could be taken to encompass any user-visible change.  That definition makes the term useless.  I would limit the term to things that make it easier for users to do things that were already reasonably possible.  And often a fairly small, incremental changes and perhaps user interface-related.  Otherwise I think of them as new features.  It's a fuzzy line between them.  "Theory Transfer Support" sounds like a new feature, AFAIK there's nothing like that now.

Reply:
I agree, we should categorize these improvements better.